src/plugins/profiling/server/routes/flamechart.ts

```diff
@@ -6,8 +6,8 @@
  * Side Public License, v 1.
  */
 import { schema } from '@kbn/config-schema';
-import type { ElasticsearchClient, IRouter, Logger } from 'kibana/server';
-import type { DataRequestHandlerContext } from '../../../data/server';
+import type { ElasticsearchClient, IRouter, Logger } from '@kbn/core/server';
```

Comment:
IntelliJ changed these two lines automatically (no idea what that syntax means).
Reverting to the left side gave me these from the git pre-commit hook:
```
      /home/tim/go/src/github.com/elastic/kibana/src/plugins/profiling/server/routes/flamechart.ts
         9:59  error  Use import request [@kbn/core/server]         @kbn/imports/uniform_imports
        10:48  error  Use import request [@kbn/data-plugin/server]  @kbn/imports/uniform_imports
      
      ✖ 2 problems (2 errors, 0 warnings)
        2 errors and 0 warnings potentially fixable with the `--fix` option.
```

Reply:
The `@...` syntax is an alias so that relative imports do not need to be used. However, I couldn't find another example in Kibana in which a plugin or application uses the suggested aliases instead of the original imports.

Could you point me to the pre-commit hook that runs? I'm curious if I can replicate the errors.

Reply:
Hm, it runs every time I do `git commit` in the kibana repo, so it should be `.git/hooks/pre-commit`. Hm, since it's aut-generated:
```
#!/usr/bin/env bash
#
# ** THIS IS AN AUTO-GENERATED FILE **
# ** PLEASE DO NOT CHANGE IT MANUALLY **
#
# GENERATED BY `node scripts/register_git_hook`
# IF YOU WANNA CHANGE SOMETHING IN THIS SCRIPT
# PLEASE RE-RUN 'yarn kbn bootstrap' or 'node scripts/register_git_hook'

# pre-commit script takes zero arguments: https://git-scm.com/docs/githooks#_pre_commit

set -euo pipefail

# Make it possible to terminate pre commit hook
# using ctrl-c so nothing else would happen or be
# sent to the output.
#
# The correct exit code on that situation
# according the linux documentation project is 130
# https://www.tldp.org/LDP/abs/html/exitcodes.html
trap "exit 130" INT

has_node() {
  command -v node >/dev/null 2>&1
}

has_nvm() {
  command -v nvm >/dev/null 2>&1
}

try_load_node_from_nvm_paths () {
  # If nvm is not loaded, load it
  has_node || {
    NVM_SH="/home/tim/.nvm/nvm.sh"

    if [ "linux" == "darwin" ] && [ -s "$(brew --prefix nvm)/nvm.sh" ]; then
      NVM_SH="$(brew --prefix nvm)/nvm.sh"
    fi

    export NVM_DIR="/home/tim/.nvm"

    [ -s "$NVM_SH" ] && . "$NVM_SH"

    # If nvm has been loaded correctly, use project .nvmrc
    has_nvm && nvm use
  }
}

extend_user_path() {
  if [ "linux" == "win32" ]; then
    export PATH="$PATH:/c/Program Files/nodejs"
  else
    export PATH="$PATH:/usr/local/bin:/usr/local"
    try_load_node_from_nvm_paths
  fi
}

# Extend path with common path locations for node
# in order to make the hook working on git GUI apps
extend_user_path

# Check if we have node js bin in path
has_node || {
  echo "Can't found node bin in the PATH. Please update the PATH to proceed."
  echo "If your PATH already has the node bin, maybe you are using some git GUI app."
  echo "Can't found node bin in the PATH. Please update the PATH to proceed."
  echo "If your PATH already has the node bin, maybe you are using some git GUI app not launched from the shell."
  echo "In order to proceed, you need to config the PATH used by the application that are launching your git GUI app."
  echo "If you are running macOS, you can do that using:"
  echo "'sudo launchctl config user path /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'"

  exit 1
}

execute_precommit_hook() {
  node scripts/precommit_hook || return 1

  PRECOMMIT_FILE="./.git/hooks/pre-commit.local"
  if [ -x "${PRECOMMIT_FILE}" ]; then
    echo "Executing local precommit hook found in ${PRECOMMIT_FILE}"
    "$PRECOMMIT_FILE" || return 1
  fi
}

execute_precommit_hook || {
  echo "Pre-commit hook failed (add --no-verify to bypass)";
  echo '  For eslint failures you can try running `node scripts/precommit_hook --fix`';
  exit 1;
}

exit 0
```

Reply:
I couldn't get the precommit hook working, but it looks like it runs eslint so I ran the following manually, ignoring the Pixi flamechart code since it was noisy:

```
yarn run lint:es --ignore-pattern "*PixiFlameChart*" src/plugins/profiling
```

Unfortunately I didn't see the errors related to '@kbn/core/server' and '@kbn/data-plugin/server'.

I didn't see any usage of either alias in the Kibana source code. Would it be possible to skip this change?

Reply:
Yeah, reverted. The linter also doesn't trigger here with 8.1. Likely, with 8.3 there are new rules for ESLint.